tests/dummy/app/pods/docs/guides/body/row-selection/template.md

```diff
@@ -87,47 +67,29 @@ itself.
-          @selectingChildrenSelectsParent={{selectingChildrenSelectsParent}}
-
-          @onSelect={{action (mut demoSelection)}}
-          @selection={{demoSelection}}
-        />
-      </EmberTable>
-    </div>
-    <div class="demo-options-group">
-      <h4>Current selection</h4>
-      <div class="demo-current-selection">{{currentSelection}}</div>
-    </div>
-    <div class="demo-options-group">
-      <h4>rowSelectionMode</h4>
-      <label> <RadioButton @name='row-selection-mode' @value='multiple' @groupValue={{rowSelectionMode}} /> multiple </label>
-      <label> <RadioButton @name='row-selection-mode' @value='single' @groupValue={{rowSelectionMode}} /> single </label>
-      <label> <RadioButton @name='row-selection-mode' @value='none' @groupValue={{rowSelectionMode}} /> none </label>
-    </div>
-    <div class="demo-options-group">
-      <h4>checkboxSelectionMode</h4>
-      <label> <RadioButton @name='checkbox-selection-mode' @value='multiple' @groupValue={{checkboxSelectionMode}} /> multiple </label>
-      <label> <RadioButton @name='checkbox-selection-mode' @value='single' @groupValue={{checkboxSelectionMode}} /> single </label>
-      <label> <RadioButton @name='checkbox-selection-mode' @value='none' @groupValue={{checkboxSelectionMode}} /> none </label>
-    </div>
-    <div class="demo-options-group">
-      <h4>selectingChildrenSelectsParent</h4>
-      <label> {{input type="checkbox" checked=selectingChildrenSelectsParent}} </label>
-    </div>
-
-    {{! END-SNIPPET }}
+    {{docs/guides/body/row-selection/examples/selection-modes
+      rowsWithChildren=rowsWithChildren
+      columns=columns
+      rowSelectionMode=rowSelectionMode
+      checkboxSelectionMode=checkboxSelectionMode
+      selectingChildrenSelectsParent=selectingChildrenSelectsParent
+      demoSelection=demoSelection
+      currentSelection=currentSelection}}
   {{/demo.example}}
 
   {{demo.snippet name='docs-example-selection-modes.hbs'}}
   {{demo.snippet label='component.js' name='docs-example-selection-modes.js'}}
 {{/docs-demo}}
+
+## Aborting a Selection
+
+Row selection follows a <a href="https://embermap.com/topics/component-side-effects/data-down-actions-up">DDAU</a> pattern, whereby the `onSelect` action handler supplied to Ember Table has control over which rows become selected. To ignore a user selection, it suffices to simply do nothing in the action handler.
+
+There is, however, some internal state that needs to be reset to fully abort a user selection. For example, Ember Table tracks the _last selected_ row in order to determine the range of rows affected in a user multi-selection. If the intent is to completely prevent a user selection, this value must not change when the action is aborted. Otherwise, a subsequent user multi-selection may target the wrong rows.
+
+To reset all internal state relating to an attempted user selection, call the `abort` function in the options object passed to the `onChange` action handler:
```

Comment:
I think from Ember Table's perspective, there's no reason to perform any cleanup at all. Selection is purely DDAU, so ET can't infer very much when it receives a new `selection` attr. The notion of a forbidden or cancelled selection is something we've made up in the application layer.

My intent with this PR was to provide what's needed to implement this logic in the application, but not force it on anyone, or change how existing applications use the `onSelect` action. I've kept the `abort` handler rather vague in its description so that we can tweak exactly what it does in the future.